app/src/layouts/DashboardLayout/MenuHeader/PremiumPlanBadge/PremiumPlanBadge.tsx

```diff
@@ -60,46 +71,57 @@ const PremiumPlanBadge = () => {
       });
   }, [activeWorkspaceId]);
 
-  try {
-    const planEndDate = new Date(planEndDateString);
-    const currentDate = new Date();
-    // @ts-ignore
-    let diffTime: any = planEndDate - currentDate;
-    daysLeft = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
-  } catch (err) {
-    Logger.log(err);
-  }
-
   if (
-    !isAppSumoDeal &&
-    planId &&
-    [APP_CONSTANTS.SUBSCRIPTION_STATUS.TRIALING, APP_CONSTANTS.SUBSCRIPTION_STATUS.CANCELLED].includes(planStatus)
+    isAppSumoDeal ||
+    [APP_CONSTANTS.SUBSCRIPTION_STATUS.TRIALING, APP_CONSTANTS.SUBSCRIPTION_STATUS.CANCELLED].includes(planStatus) ||
+    daysLeft > 30 ||
+    !isUserLoggedIn
   ) {
-    if (daysLeft > 30) {
-      return null;
-    }
+    return null;
+  }
```

Comment:
Why has this condition been reversed?

I can see a lot of cases are missing. 
1. the badge would never be visible in when status is trialing.

Can we make only necessary UI changes in this PR without touching any of the rendering logic? That way it would easy to review and test. Pricing and plan logic are quite old and its quite difficult to review and test when the blast radius is increased by changing these logics.